### PR TITLE
Milestone 8

### LINKED ISSUE #60: Add S3-proxy deployment to loclaega chart

### PR DESCRIPTION
**Pull request by jbygdell**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Changes made:

1) Adds DOA as main data out component
2) Adds S3inbox as inbox option
3) Don't deploy data-api by default
4) Updates e2e-tester to work with c4gh 1.0

### Related issues:

Fixes #60 
### Additional information:

There are some related pending PRs in:
~~neicnordic/LocalEGA#19~~
~~neicnordic/LocalEGA-tester#35~~
~~neicnordic/LocalEGA-deploy-init#38~~

### CODE REVIEW

**Comment by blankdots** [comment]
localega charts worth a version bump :)

File: README.md [hunk 1]
```diff
@@ -112,12 +112,15 @@ Parameter | Description | Default
 `config.postgres_in_user` | Database username for ingestion | `lega_in`
 `config.postgres_out_user` | Database username for outgestion | `lega_out`
 `config.data_storage_type` | Backend storage type, `S3Storage` or `FileStorage` | `S3Storage`
-`config.data_storage_url` | URL to S3 storage instance | `""`
-`config.data_storage_s3_bucket` | S3 storage bucket | `""`
-`config.data_storage_s3_region` | S3 storage region | `""`
-`config.data_storage_s3_chunk_size` | S3 chunk size in MB | `16`
+`config.s3_archive_url` | URL to S3 archive instance | `""`
+`config.s3_archive_bucket` | S3 archive bucket | `""`
+`config.s3_archive_region` | S3 archve region | `""`
```
Comment (blankdots) [inline README.md]:
```suggestion
`config.s3_archive_region` | S3 archive region | `""`
```

File: ega-charts/localega/templates/doa-deploy.yaml [hunk 2]
```diff
@@ -89,7 +89,7 @@ spec:
         - name: CERT_KEY
           value: "/ega/tls/doa.ca.key.der"
         - name: CRYPT4GH_PRIVATE_KEY_PATH
-          value: "/etc/ega/c4gh/key.1.c4gh.sec"
+          value: "/etc/ega/c4gh/ega_key.c4gh.sec"
```
Comment (blankdots) [inline ega-charts/localega/templates/doa-deploy.yaml:92]: this is more of a question, I understand using it like this for e2e-tester, but why for DOA ?
wouldn't it be a bit better to have in a secret, I mean the `lega-secret` it is already there
the same goes for the key.pass below.

File: ega-charts/localega/templates/verify-conf.yaml [hunk 3]
```diff
@@ -30,6 +30,10 @@ data:
     cacertfile = /etc/ega/ssl/{{ .Values.config.tls_ca_root_file }}
     certfile = /etc/ega/ssl/verify{{ .Values.config.tls_cert_ending }}
     keyfile = /etc/ega/ssl/verify{{ .Values.config.tls_key_ending }}
+    [c4gh_file]
+    loader_class = C4GHFileKey
+    passphrase = {{ .Values.secrets.ega_c4gh_passphrase }}
+    filepath = /etc/ega/c4gh/ega_key.c4gh.sec
```
Comment (blankdots) [inline ega-charts/localega/templates/verify-conf.yaml:36]: in the long term maybe have this as a parameter as well. We do have it like that for other values: certs, keys etc.

**Review by blankdots — CHANGES_REQUESTED** [review]
would be nice not to have some paths predefined. In rest LGTM.

File: ega-charts/localega/templates/doa-deploy.yaml (hunk 2 above)
Comment (jbygdell) [reply ega-charts/localega/templates/doa-deploy.yaml:92]: It is in a secret, that's just the path tho the file to read.

File: ega-charts/localega/templates/verify-conf.yaml (hunk 3 above)
Comment (jbygdell) [reply ega-charts/localega/templates/verify-conf.yaml:36]: The file name at least, the path is fixed from the mount point of the secret.

**Review by blankdots — DISMISSED** [review]

**Review by blankdots — DISMISSED** [review]
LGTM, we will need different github actions to test combinations of sftp inbox or S3 Inbox and FileStorage or S3Storage, but that can be a different issue.

File: ega-charts/localega/templates/s3inbox-config.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+{{- end }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "localega.name" . }}-s3inbox-users
+type: Opaque
+data:
+  users.csv: |-
+    {{ printf "%s,%s"  .Values.secrets.s3_inbox_user_access_key .Values.secrets.s3_inbox_user_secret_key | b64enc}}
+{{- end }}
```
Comment (viklund) [inline ega-charts/localega/templates/s3inbox-config.yaml]: No new-line

File: ega-charts/localega/templates/s3inbox-svc.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    app: {{ template "localega.name" . }}-s3inbox
+spec:
+  ports:
+  - name: 
+    port: {{ .Values.s3inbox.servicePort }}
+    targetPort: {{ .Values.s3inbox.port }}
+    protocol: TCP
+  selector:
+    app: {{ template "localega.name" . }}-s3inbox
+{{- end }}
```
Comment (viklund) [inline ega-charts/localega/templates/s3inbox-svc.yaml]: No new-line

**Review by viklund — CHANGES_REQUESTED** [review]
The following files don't end with a newline character:

- ega-charts/localega/templates/s3inbox-config.yaml
- ega-charts/localega/templates/s3inbox-svc.yaml

**Review by blankdots — DISMISSED** [review]

**Review by blankdots — APPROVED** [review]

**Review by viklund — APPROVED** [review]

**Pull request merged**